### PR TITLE
`styled_component_base` as lower level primitive of `styled_component`

### PR DESCRIPTION
**Pull request by WorldSEnder**

### CODE REVIEW

File: packages/stylist/src/yew/mod.rs [hunk 1]
```diff
@@ -3,7 +3,11 @@
 use yew::html::Classes;
 use yew::html::IntoPropValue;
 
-/// A procedural macro to style Yew component.
+/// A procedural macro to style a function component. Specifically this introduces a
```
Comment (futursolo) [inline packages/stylist/src/yew/mod.rs:6]: Better to move the sentence after the fist full stop to a new paragraph? So it looks nicer in the documentation of `stylist::yew`.

**Review by futursolo — CHANGES_REQUESTED** [review]
Maybe `impl_styled_component` is a better name than `styled_component_base` for this macro?

```rust
#[impl_styled_component]
#[function_component(Comp)]
fn comp() -> Html {
    ...
}
```

**Comment by WorldSEnder** [comment]
> Maybe `impl_styled_component` is a better name than `styled_component_base` for this macro?

I was also debating `use_stylist` as I think the angle of using it in custom hooks is a valid and useful use case. The `use_style!` macro as is invokes `use_context` once per usage, while the injected `css!` macro reuses the same manager, retrieved only once. Quite unsure if there's any strong conclusions to draw from this.

**Comment by futursolo** [comment]
> use_style! macro as is invokes use_context once per usage

I think there's very little incentive to optimise this.

I do not think a component would have many `use_style!` hooks in the same component to the extent that the performance would be degraded.

It is common to access the same context with `use_context` multiple times in custom hooks in React.
Accessing context content should be a cheap operation, or else it's a problem of Yew to optimise.

**Review by futursolo — APPROVED** [review]

**Pull request merged**